core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java

```diff
@@ -84,6 +88,7 @@ private void terminateMetaDataInstances(Stack stack) {
             long timeInMillis = Calendar.getInstance().getTimeInMillis();
             metaData.setTerminationDate(timeInMillis);
             metaData.setInstanceStatus(InstanceStatus.TERMINATED);
+            instanceMetaDataRepository.save(metaData);
```

Comment:
Instead of saving it in each loop can't we save once for all with save(stack.getRunningInstanceMetaData())?